gslib/gcs_json_credentials.py

```diff
@@ -73,6 +77,75 @@
 GOOGLE_OAUTH2_DEFAULT_FILE_PASSWORD = 'notasecret'
 
 
+class PKCS12Signer(crypt_base.Signer, crypt_base.FromServiceAccountMixin):
+  """Signer for a p12 service account key."""
+
+  def __init__(self, key):
+    self._key = key
+
+  # Defined in crypt_base.Signer interface.
+  # It is not used by p12 service account keys.
+  @property
+  def key_id(self):
+    return None
+
+  def sign(self, message):
+    message = _helpers.to_bytes(message)
+    from google.auth.crypt import _cryptography_rsa  # pylint: disable=g-import-not-at-top
+    return self._key.sign(
+        message,
+        _cryptography_rsa._PADDING,  # pylint: disable=protected-access
+        _cryptography_rsa._SHA256)  # pylint: disable=protected-access
+
+  @classmethod
+  def from_string(cls, key_strings, key_id=None):
+    del key_id
+    key_string, password = (_helpers.to_bytes(k) for k in key_strings)
+    from cryptography.hazmat.primitives.serialization import pkcs12  # pylint: disable=g-import-not-at-top
```

Comment:
What's the reason for not importing it at the top, can you please add a comment explaining the reason?

Reply:
The reason behind lazy import is inspired from:
1. Cryptography package is not bundled with gsutil, keeping it at top would throw error - ```module not found```.
2. Since this module was only used by this path, importing it here makes it more readable to developers.
3.  go/rapid-python-performance#binary-start-up-time

I've added the comment